### PR TITLE
Feat/payment update

### PR DESCRIPTION
**Pull request by ProggerOleg**
## Describe your changes
- I added endpoint to withdraw and top-up user's balance
Screenshot from 2023-12-29 11-24-09

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [ ] use ConfigService instead of process.env
- [ ] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

### CODE REVIEW

**Review by Lilly-nn — APPROVED** [review]

**Review by fedasevich — APPROVED** [review]

File: src/modules/payment/payment.controller.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    status: HttpStatus.INTERNAL_SERVER_ERROR,
+    description: ErrorMessage.InternalServerError,
+  })
+  @Post(PaymentAPiPath.TopUp)
+  async topUpBalance(
+    @Body() { balance }: { balance: number },
+    @Req() request: AuthenticatedRequest,
+  ): Promise<void> {
+    const userId = request.user.id;
+    if (!userId) {
```
Comment (Simpsonkk) [inline src/modules/payment/payment.controller.ts]: why do we need two identical endpoints?

File: src/modules/payment/payment.service.ts [hunk 2]
```diff
@@ -344,4 +344,17 @@ export class PaymentService {
...
+  ): Promise<void> {
+    try {
+      await this.userService.updateUserInfo(userId, {
+        balance: updatedBalance,
+      });
+    } catch (error) {
+      throw new HttpException(error.message, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
 }
```
Comment (Simpsonkk) [inline src/modules/payment/payment.service.ts:360]: why do we need this method if we have updateUserInfo? You can use it.

File: src/modules/payment/payment.controller.ts
Comment (Simpsonkk) [inline src/modules/payment/payment.controller.ts:1]: as for me, we don`t need the payment.controller.ts at all. You can use user.controller.ts ->Patch updateProfile and update any User table column.

Comment (ProggerOleg) [reply src/modules/payment/payment.controller.ts:1]: All operations connected to balance I would prefer to see in payment module, also later we will add here functionality of creation a record about top-up/withdraw, so i think it should stay

File: src/modules/payment/payment.service.ts (hunk 2 above)
Comment (ProggerOleg) [reply src/modules/payment/payment.service.ts:360]: Later will update this method to create record in transations table which won't be comfortable to make in userService

File: src/modules/payment/payment.controller.ts (hunk 1 above)
Comment (ProggerOleg) [reply src/modules/payment/payment.controller.ts]: Ok, agree, fixed

**Pull request merged**